ntoskrnl/io/iomgr/deviface.c

```diff
@@ -304,7 +305,189 @@ IoGetDeviceInterfaceAlias(IN PUNICODE_STRING SymbolicLinkName,
+    if (!NT_SUCCESS(Status))
+    {
+        DPRINT1("RtlStringFromGUID() failed with status 0x%08lx\n", Status);
+        goto cleanup;
+    }
+
+    BaseKeyName.Length = 0;
+    BaseKeyName.MaximumLength += GuidString.Length;
+    BaseKeyName.Buffer = ExAllocatePoolWithTag(NonPagedPool,
+                                               BaseKeyName.MaximumLength,
+                                               TAG_IO);
+    if (!BaseKeyName.Buffer)
+    {
+        DPRINT1("ExAllocatePoolWithTag() failed\n");
+        Status = STATUS_INSUFFICIENT_RESOURCES;
+        goto cleanup;
+    }
+
+    /* Concatenate alias GUID string with base key */
+    Status = RtlAppendUnicodeStringToString(&BaseKeyName, &GuidString);
+    if (!NT_SUCCESS(Status))
+    {
+        DPRINT1("RtlAppendUnicodeStringToString() failed with status 0x%08lx\n", Status);
+        goto cleanup;
+    }
+
+    /* Open base registry key */
+    Status = IoOpenDeviceInterfaceRegistryKey(&BaseKeyName,
+                                              GENERIC_READ,
+                                              &InterfaceKey);
+    if (!NT_SUCCESS(Status))
+    {
+        DPRINT1("IoOpenDeviceInterfaceRegistryKey() failed with status 0x%08lx\n", Status);
+        goto cleanup;
+    }
+
+    Status = ZwQueryKey(InterfaceKey,
+                        KeyFullInformation,
+                        NULL,
+                        0,
+                        &Size);
+    if (!NT_SUCCESS(Status) && Status != STATUS_BUFFER_TOO_SMALL)
+    {
+        DPRINT1("ZwQueryKey() failed with status 0x%08lx\n", Status);
+        goto cleanup;
+    }
+
+    fip = ExAllocatePoolWithTag(NonPagedPool,
+                                Size,
+                                TAG_IO);
+    if (!fip)
+    {
+        DPRINT1("ExAllocatePoolWithTag() failed\n");
+        Status = STATUS_INSUFFICIENT_RESOURCES;
+        goto cleanup;
+    }
+
+    Status = ZwQueryKey(InterfaceKey,
+                        KeyFullInformation,
+                        fip,
+                        Size,
+                        &Size);
+    if (!NT_SUCCESS(Status))
+    {
+        DPRINT1("ZwQueryKey() failed with status 0x%08lx\n", Status);
+        goto cleanup;
+    }
+
+    /* Enumerate subkeys (i.e. the different device objects) */
+    for (i = 0; i < fip->SubKeys; i++)
+    {
+        Status = ZwEnumerateKey(InterfaceKey,
+                                i,
+                                KeyBasicInformation,
+                                NULL,
+                                0,
+                                &Size);
+        if (Status == STATUS_NO_MORE_ENTRIES)
+        {
+            break;
+        }
+        else if (!NT_SUCCESS(Status) && Status != STATUS_BUFFER_TOO_SMALL)
+        {
+            DPRINT1("ZwEnumerateKey() failed with status 0x%08lx\n", Status);
+            goto cleanup;
+        }
+
+        bip = ExAllocatePoolWithTag(NonPagedPool,
```

Comment:
Using `PagedPool` for "bip" should be fine, since your function runs in PAGED_MODE and this "bip" is used locally there only.

Reply:
Fixed.